tests/functional/test.cli.sign.js

```diff
@@ -48,4 +51,95 @@ describe('web-ext sign', () => {
        });
      })
   );
+
+  it('should use config file if required parameters are not in the arguments',
+     () => withTempAddonDir({addonPath: minimalAddonPath}, (srcDir, tmpDir) => {
+       fs.writeFileSync(
+         path.join(tmpDir, 'web-ext-config.js'),
+         `
+module.exports = {
+  sign: {
+    apiKey: "FAKEAPIKEY",
+    apiSecret: "FAKEAPISECRET",
+  },
+};`
+       );
+
+       // Optional properties goes here, so that the above string can be used as
+       // documentation on how to configure the key and secret through a file.
+       fs.writeFileSync(
+         path.join(tmpDir, 'package.json'),
+         `
+         {
+           "webExt": {
+             "sign": {
+               "apiUrlPrefix": "http://localhost:8989/fake/api/v3"
+             },
+             "sourceDir": ${JSON.stringify(srcDir)}
+           }
+         }
```

Comment:
I've intentionally kept the value as a string (especially the `web-ext-config.js` file), so that we can refer to it as documentation (an example of how to write the content). The package.json config content was separated, to prevent users from blindly copying the `apiUrlPrefix` and `sourceDir` parameters.

Given this explanation, do you still want me to use `JSON.stringify` on the minimal object?

Reply:
I'm not 100% sure that I get what you mean with "The package.json config content was separated, to prevent users from blindly copying the apiUrlPrefix and sourceDir parameters", in which context you would like to use it and who would be the documentation targeting. 

If it is something that we want to highlight and be able to refer to, it may be good to actually document it properly in a PR to the extension-workshop repo. The tests may be rewritten, refactored and changed over the time and so even if that had originally a particular purpose (which wasn't testing-related) we may actually forget about it at some point and change it in a unrelated PR. 

And so I think that I would still prefer it (but in the end it is still just a nit, not something that I would actual block this on).

Reply:
Done. I don't care that much about using `JSON.stringify` with package.json.

Documentation in the extension-workshop repo would be ideal, but until that happens the only form of documentation is this repo. It is easier to point users to a git link to this test than to allocate time to write proper documentation for this feature. The original issue that this PR fixes showed the "wrong" way to use this feature, hence my desire to have one copy-pasteable template, somewhere.